services/publish/src/handlers/publish.rs

```diff
@@ -29,6 +30,18 @@ async fn do_publish<R: RedisOperations>(
 
     let stream_key = StreamKey::new(org_id, channel_id);
 
+    // Track stream activity BEFORE publish to prevent orphaned streams.
+    redis
+        .track_stream_update(&stream_key, Utc::now().timestamp())
```

Comment:
not sure i'm reading this right but is this actually updating the sorted set on *every* publish? do we want to avoid that many updates?

Reply:
The publish service runs multiple stateless pods, so an in-memory cache wouldn't coordinate across instances. The same stream's publishes could hit different pods leading to stale timestamps in Redis. While this would reduce Redis calls, `ZADD` is sub-millisecond. We're trading correctness guarantees for an optimization we haven't proven we need.
I'd prefer to keep it the simple, correct implementation unless we measure that `ZADD` is actually a bottleneck. If it becomes one, we could consider application-level sharding of the sorted set rather than caching to maintain correctness guarantees while distributing the load.